mcstatus/tests/test_motd.py

```diff
@@ -0,0 +1,239 @@
+                WebColor(hex="#e0b2e1", rgb=(224, 178, 225)), Formatting.BOLD, "F", Formatting.RESET,
+                WebColor(hex="#efa8ea", rgb=(239, 168, 234)), Formatting.BOLD, "u", Formatting.RESET,
+                WebColor(hex="#ff9ff4", rgb=(255, 159, 244)), Formatting.BOLD, "n ", Formatting.RESET,
+                WebColor(hex="#b3eeff", rgb=(179, 238, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                WebColor(hex="#b9ecff", rgb=(185, 236, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                WebColor(hex="#c0eaff", rgb=(192, 234, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                WebColor(hex="#c7e8ff", rgb=(199, 232, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                WebColor(hex="#cee6ff", rgb=(206, 230, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                WebColor(hex="#d5e4ff", rgb=(213, 228, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                WebColor(hex="#dce2ff", rgb=(220, 226, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                WebColor(hex="#e3e0ff", rgb=(227, 224, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                WebColor(hex="#eadeff", rgb=(234, 222, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                WebColor(hex="#f1dcff", rgb=(241, 220, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                WebColor(hex="#f8daff", rgb=(248, 218, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                WebColor(hex="#ffd9ff", rgb=(255, 217, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                WebColor(hex="#f4dcff", rgb=(244, 220, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                WebColor(hex="#f9daff", rgb=(249, 218, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                WebColor(hex="#ffd9ff", rgb=(255, 217, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                MinecraftColor.WHITE, " \n ",
+                WebColor(hex="#E5E5E5", rgb=(229, 229, 229)), Formatting.BOLD,
+                "The server has been updated to ", Formatting.RESET,
+                WebColor(hex="#97ABFF", rgb=(151, 171, 255)), Formatting.BOLD, "1.17.1", Formatting.RESET,
+                "",
+                # fmt: on
+            ]
+        )
+
+    @pytest.mark.parametrize("bedrock", (True, False))
+    def test_bedrock_parameter_nothing_changes(self, bedrock: bool):
+        assert Motd.parse([{"color": "minecoin_gold", "text": " "}], bedrock=bedrock).raw == [
+            MinecraftColor.MINECOIN_GOLD,
+            " ",
+        ]
+
+    @pytest.mark.parametrize("bedrock", (True, False))
+    def test_parse_as_str_ignore_minecoin_gold_on_java(self, bedrock: bool):
+        assert Motd.parse("&g", bedrock=bedrock).raw == [MinecraftColor.MINECOIN_GOLD if bedrock else "&g"]
```

Comment:
This should use parametrize with two values, first being the bedrock bool, and second being the expected result. We shouldn't need to check the parametrized bedrock bool and determine the result based on it from within the actual logic.

Reply:
Fixed in [last force-push](https://github.com/py-mine/mcstatus/compare/9618efbbeca07cf4b99f5f834fa8b420969fd017..bccee991168674292f251d33cc2c2f19584abfb9).